lib/trello/card.rb

```diff
@@ -491,31 +481,49 @@ def fields_has_key?(fields, key)
-      attributes[:due]                    = Time.iso8601(fields[SYMBOL_TO_STRING[:due]]) rescue nil if fields.has_key?(SYMBOL_TO_STRING[:due])
-      attributes[:due]                    = fields[:due] if fields.has_key?(:due)
-      attributes[:due_complete]           = fields[SYMBOL_TO_STRING[:due_complete]] if fields.has_key?(SYMBOL_TO_STRING[:due_complete])
-      attributes[:due_complete]           ||= false
-      attributes[:closed]                 = fields[SYMBOL_TO_STRING[:closed]] if fields.has_key?(SYMBOL_TO_STRING[:closed])
-      attributes[:url]                    = fields[SYMBOL_TO_STRING[:url]] || attributes[:url]
-      attributes[:short_url]              = fields[SYMBOL_TO_STRING[:short_url]] || attributes[:short_url]
-      attributes[:board_id]               = fields[SYMBOL_TO_STRING[:board_id]] || attributes[:board_id]
-      attributes[:member_ids]             = fields[SYMBOL_TO_STRING[:member_ids]] || fields[:member_ids] || attributes[:member_ids]
-      attributes[:list_id]                = fields[SYMBOL_TO_STRING[:list_id]] || fields[:list_id] || attributes[:list_id]
-      attributes[:pos]                    = fields[SYMBOL_TO_STRING[:pos]] || fields[:pos] || attributes[:pos]
-      attributes[:labels]                 = (fields[SYMBOL_TO_STRING[:labels]] || []).map { |lbl| Trello::Label.new(lbl) }.presence || attributes[:labels].presence || []
-      attributes[:card_labels]            = fields[SYMBOL_TO_STRING[:card_labels]] || fields[:card_labels] || attributes[:card_labels]
-      attributes[:last_activity_date]     = Time.iso8601(fields[SYMBOL_TO_STRING[:last_activity_date]]) rescue nil if fields.has_key?(SYMBOL_TO_STRING[:last_activity_date])
-      attributes[:cover_image_id]         = fields[SYMBOL_TO_STRING[:cover_image_id]] || attributes[:cover_image_id]
-      attributes[:badges]                 = fields[SYMBOL_TO_STRING[:badges]] || attributes[:badges]
-      attributes[:card_members]           = fields[SYMBOL_TO_STRING[:card_members]] || attributes[:card_members]
-      attributes[:source_card_id]         = fields[SYMBOL_TO_STRING[:source_card_id]] || fields[:source_card_id] || attributes[:source_card_id]
-      attributes[:source_card_properties] = fields[SYMBOL_TO_STRING[:source_card_properties]] || fields[:source_card_properties] || attributes[:source_card_properties]
+      %i[
+        id short_id url short_url last_activity_date
+        badges card_members
+      ].each do |attr_key|
+        attributes[attr_key] = parse_readonly_fields(fields, attr_key)
+      end
+
+      %i[
+        name desc due due_complete closed board_id
+        member_ids list_id pos labels card_labels
+        cover_image_id source_card_id source_card_properties
+      ].each do |attr_key|
+        attributes[attr_key] = parse_writable_fields(fields, attr_key)
+      end
+
+      attributes[:last_activity_date] = Time.iso8601(attributes[:last_activity_date]) rescue nil if attributes[:last_activity_date]
```

Comment:
it is a bit hard to read the time casting, what do you think about having a method for that responsibility?

Reply:
Plus, what do you think about removing the presence check for `attributes[:last_activity_date]`? Isn't that `rescue nil` would handle the case `Time.iso8601(nil)`?

Reply:
I will do this